builtin/providers/aws/structure.go

```diff
@@ -1567,3 +1567,24 @@ func flattenPolicyAttributes(list []*elb.PolicyAttributeDescription) []interface
 
 	return attributes
 }
+
+// Takes a value containing JSON string and passes it through
+// the JSON parser to normalize it, returns either a parsing
+// error or normalized JSON string.
+func normalizeJsonString(jsonString interface{}) (string, error) {
+	var j interface{}
+
+	if jsonString == nil || jsonString.(string) == "" {
+		return "", nil
+	}
+
+	s := jsonString.(string)
+
+	err := json.Unmarshal([]byte(s), &j)
+	if err != nil {
+		return s, err
+	}
+
+	bytes, _ := json.Marshal(j)
```

Comment:
Indeed, this is a copy-and-paste. I can add error checking, although the only thing that could upset things here would be some odd thing making memory backing this corrupted.

Having said that, things like e.g. (errcheck)[https://github.com/kisielk/errcheck] could mark this as a potential issue.